polaris-react/src/components/Tabs/Tabs.scss

```diff
@@ -356,7 +356,7 @@
     }
 
     &:not([aria-disabled='true']):hover {
-      background-color: var(--p-color-bg-secondary-experimental);
+      background-color: var(--p-color-bg-transparent-hover-experimental);
```

Comment:
@bernardojoaogarcia Initially left this out, as unsure if `DisclosureTab` is different on purpose. (couldnt' find the relevant asset in Figma) 
I've made this change though, as the pre-uplift styles use `bg-hover` same as the hover styles of standard tabs.